WordPress/gutenberg
# EditPostPreferencesModal: refactor to remove snapshots for index.js

## What?
Every second or third run, I'm seeing the following errors in both local and CI JS unit tests

```console
FAIL packages/edit-post/src/components/preferences-modal/test/index.js (5.323 s)
  ● EditPostPreferencesModal › should match snapshot when the modal is active › large viewports

    expect(received).toMatchSnapshot()

    Snapshot name: `EditPostPreferencesModal should match snapshot when the modal is active large viewports 1`

    - Snapshot  - 0
    + Received  + 2
```


This PR already tried to fix this test:

- https://github.com/WordPress/gutenberg/pull/53814

Historical context: https://github.com/WordPress/gutenberg/pull/52133#issuecomment-1680399985

Instead of waiting around for aria-kit to add the tab-index attribute, this PR just gets rid of the relevant snapshot, and, rather, tests for main view elements to verify that the viewport switch is made. Thanks to @andrewserong for helping to debug

The snapshots seem to only test whether `useViewportMatch` renders the correct views. By testing for some key elements we don't need the entire snapshot.

There's also a strong argument to turn this into an E2E test, but for now, I'm just getting the tests to pass to unblock folks.


## How are you?
All good here, thanks!

## Testing Instructions
Check CI

`npm run test:unit packages/edit-post/src/components/preferences-modal/test/index.js` until you get sick of it
